### DISCUSSION TITLE
Relacionamento com campo string

### DISCUSSION THREAD

**Discussion by almp1** [balta-io/forum#155] [Tira Dúvidas] [answered]
Olá Balta, td bem?

Tenho uma necessidade de relacionar 2 tabelas  one to many com campo string que não são chaves das tabelas

    public class  Schedule
    {
        public int Id { get; set; }
        public string Comment { get; set; }
        public string Name { get; set; }
    }

    public class  Products
    {
        public int Id { get; set; }
        public string Product { get; set; }
    }

Preciso relacionar Schedule.Name com Products.Product

Como posso fazer isso de forma simples

Estou usando o Entity 7.0

Obrigado

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde, @almp1... neste caso não é um "relacionamento"... basta fazer um select nos produtos onde schedule.name seja igual ao produto. 

Caso precise selecionar vários produtos você pode usar um JOIN (Não vai ser performático por ausência das chaves)... talvez uma forma mais otimizada seria ter uma view para isto.

Dá pra fazer com LINQ também...
```csharp
var data =
    from product in context.Products
    join schedule in context.Schedules on product.Name equals schedule.Product
    select new
    {
        ...
    };
```
